nums/core/array/application.py

```diff
@@ -425,7 +425,7 @@ def diag(self, X: BlockArray) -> BlockArray:
                                                                         grid_meta,
                                                                         syskwargs=syskwargs)
         elif len(X.shape) == 2:
-            assert X.shape[0] == X.shape[1]
+            assert X.shape[0] == X.shape[1], "X must be a square matrix"
```

Comment:
Can you include punctuation? Can you also add a similar error message to the `block_shape` assertion below?

Reply:
Sure. You mean punctuation in the error message right?